include/spore/codegen/codegen_cache.hpp

```diff
@@ -47,12 +50,21 @@ namespace spore::codegen
 
         bool check_and_update(const std::string_view& file)
         {
+            std::string file_content;
+            if (!spore::codegen::read_file(file, file_content))
+            {
+                return false;
+            }
+
             std::string hash;
-            if (!spore::codegen::hash_file(file, hash))
+            if (!spore::codegen::hash_file_content(file_content, hash))
             {
                 return false;
             }
 
+            std::uintmax_t file_size = std::filesystem::file_size(file);
```

Comment:
This execute `stat` system call twice. We should stat once and retrieve the file size and the mtime from the stat structure.

Reply:
Does this really make a difference? It's a much messier API for negligible performance difference over the entire runtime of the application. 

`std::filesystem` is much more readable and explicit about what we're doing. It's also portable, while `stat` is linux specific.

Let me know if you still want to do the syscall and I'll make the change.

Reply:
Ha sorry, I mean there's an equivalent method `std::filesystem::status` that returns a `std::filesystem::file_status` that contain the size and the mtime IIRC.

Reply:
I'm looking at the structure and it doesn't look like it contain much (https://en.cppreference.com/w/cpp/filesystem/file_status). Let me know if it doesn't contain the file size etc, and I'll resolve this comment.